l5kit/l5kit/dataset/chop_dataset.py

```diff
@@ -0,0 +1,97 @@
+import argparse
+from pathlib import Path
+
+import numpy as np
+from zarr import convenience
+
+from l5kit.data import ChunkedDataset
+from l5kit.data.zarr_utils import zarr_scenes_chunk
+from l5kit.dataset.select_agents import TH_DISTANCE_AV, TH_EXTENT_RATIO, TH_YAW_DEGREE, select_agents
+from l5kit.evaluation import export_zarr_to_csv
+
+MIN_FUTURE_STEPS = 10
+FUTURE_STEPS = 50
+
+
+def create_chopped_dataset(
+    zarr_path: str, th_agent_prob: float, num_frames_to_copy: int, num_frames_gt: int, min_frame_future: int
+) -> str:
+    """
+    Create a chopped version of the zarr that can be used for evaluation. This function is used to generate tests
+    set that have no GT in the data for selected agents.
+
+    Store:
+     - a dataset where each scene has been chopped at `num_frames_to_copy` frames;
+     - a mask for agents for those final frames based on the original mask and a threshold on the future_frames;
+     - the GT csv for those agents
+
+    Args:
+        zarr_path (str): zarr path as a string
```

Comment:
Minor, but I feel these comments are a bit hastily written: 'threshold over agents prob ...', 'min number of frames available in the future for an agent to be taken'